pubsub/pulsar/pulsar.go

```diff
@@ -118,6 +118,8 @@ type Pulsar struct {
 	wg       sync.WaitGroup
 }
 
+var newPulsarClient = pulsar.NewClient
```

Comment:
is this used outside of the func? If not then it should be scoped within the func pls